### PR TITLE
WIP: Feature/moderation commands

### PR DESCRIPTION
**Pull request by MrDienns**
Pull request which introduces a command setup and some moderation commands included.

### CODE REVIEW

**Review by MisterErwin — APPROVED** [review]
Except some ToDos and format inconsistencies - my :+1:

**Comment by MrDienns** [comment]
@MisterErwin which format inconsistencies exactly? It's still a work-in-progress for some commands, but format wise it should be okay. Maybe I missed some things here and there.

I'm gonna have to look into a testing solution later, as mentioned in Discord.

**Comment by MisterErwin** [comment]
Mostly the blank lines after some if statements, but not all:p

Dennis van der Veeke <notifications@github.com> schrieb am Mi., 24. Apr.
2019, 10:57:

File: src/main/java/com/dyescape/dyescapebot/command/discord/resolver/MemberResolver.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        // popFirstArgument will always give us the correct argument string
+        String argument = context.popFirstArg();
+
+        // TODO: Improve this
+        if (!argument.startsWith("<@")) {
+            throw new InvalidCommandArgument("Please tag the user.");
+        }
+
+        // Let's get the Discord ID from the tag string
+        Pattern idPattern = Pattern.compile("<@.*?(\\d+)>");
```
Comment (2008Choco) [inline src/main/java/com/dyescape/dyescapebot/command/discord/resolver/MemberResolver.java]: Compile once. This should be a constant

File: src/main/java/com/dyescape/dyescapebot/util/TimeUtil.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+                .appendDays().appendSuffix("d")
+                .appendDays().appendSuffix("day")
+                .appendDays().appendSuffix("days")
+                .appendHours().appendSuffix("h")
+                .appendHours().appendSuffix("hour")
+                .appendHours().appendSuffix("hours")
+                .appendMinutes().appendSuffix("m")
+                .appendMinutes().appendSuffix("minute")
+                .appendMinutes().appendSuffix("minutes")
+                .toFormatter();
```
Comment (2008Choco) [inline src/main/java/com/dyescape/dyescapebot/util/TimeUtil.java]: Should also be a constant

File: src/main/java/com/dyescape/dyescapebot/util/TimeUtil.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+                } else {
+                    builder.append(String.format(", %s minutes", mins % 60));
+                }
+            } else {
+                if (mins % 60 == 1) {
+                    builder.append(String.format("%s minute", mins % 60));
+                } else {
+                    builder.append(String.format("%s minutes", mins % 60));
+                }
+            }
```
Comment (2008Choco) [inline src/main/java/com/dyescape/dyescapebot/util/TimeUtil.java:77]: I'm sure there's a cleaner way to write this. The pluralization especially. A method could be made to simplify it.

**Review by 2008Choco — CHANGES_REQUESTED** [review]

File: src/main/java/com/dyescape/dyescapebot/util/TimeUtil.java (hunk 3 above)
Comment (MrDienns) [reply src/main/java/com/dyescape/dyescapebot/util/TimeUtil.java:77]: Simplified it a bit. Got rid of the overload of unnecessairy if statements. Would a method still be nice to have with the new version?

**Review by 2008Choco — APPROVED** [review]

**Comment by MrDienns** [comment]
Closing this PR in favour of https://github.com/Dyescape/DyescapeBot/pull/15 because the underlying framework was changed.

**Pull request closed without merging**